libsql-server/src/auth/user_auth_strategies/jwt.rs

```diff
@@ -80,6 +88,20 @@ mod jwt_time {
     }
 }
 
+fn validate_any_jwt(
+    jwt_keys: &Vec<jsonwebtoken::DecodingKey>,
+    jwt: &str,
+) -> Result<Authenticated, AuthError> {
+    for (index, jwt_key) in jwt_keys.iter().enumerate() {
+        let result = validate_jwt(&jwt_key, jwt);
+        if result.is_ok() || index == jwt_keys.len() - 1 {
```

Comment:
we're swallowing individual errors
we should at least log a warning if result is Err

Reply:
The problem is that those errors are expected. If I have to different keys, only one of them can decrypt the JWT, so the other one will error out. If we log them it will get very noisy without adding value.

Reply:
maybe debug log then? to turn it on only when troubleshooting?

Reply:
After giving this more thought I believe the intent - and desired behavior - is to do our very best to communicate to a user why their request is failing. As such, would should try and avoid obfuscating the real error with a generic one if possible. My thought process is that if we receive any error other than `InvalidJwt` then it means that we were able to parse the incoming JWT with the current key but that it is either expired or immature. There shouldn't be any reason to keep trying other keys in this scenario and we should just communicate "hey, your jwt is expired" right away. Otherwise, if we go through all of the keys but all return back `InvalidJwt` then we can just return that.